test/commands/deploy/pipeline.test.ts

```diff
@@ -698,6 +702,78 @@ describe('deploy pipeline', () => {
         expect(requestArgument.body).to.contain('"changeBundleName":"DummyChangeBundleName"');
       });
 
+    describe('Processing VCS event flow', () => {
+      beforeEach(() => {
+        // mock the pipeline stage record
+        pipelineStageMock = {
+          Id: 'mock-id',
+          sf_devops__Branch__r: {
+            sf_devops__Name__c: 'mockBranchName',
+          },
+          sf_devops__Pipeline__r: {
+            sf_devops__Project__c: 'mockProjectId',
+          },
+          sf_devops__Pipeline_Stages__r: undefined,
+        };
+        fetchAndValidatePipelineStageStub = sandbox
+          .stub(Utils, 'fetchAndValidatePipelineStage')
+          .resolves(pipelineStageMock);
+      });
+      test
```

Comment:
Can we start a best practice of documenting the test cases.  Sometimes they can be difficult to grok.  It does not need to be much.  In this case you have comments in the test setup, which helps, but a summary would be even better when scanning.

//Test the case where we get 49 CONFLICT errors and on the 50th we get a different error